### PR TITLE
Feature/admin applications

### PR DESCRIPTION
**Pull request by AltynbekMamasadykov**

### CODE REVIEW

File: src/main/java/kg/airbnb/airbnb/apis/AdminApi.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+public class AdminApi {
+
+    private final AnnouncementServiceImpl announcementServiceImpl;
+
+    @GetMapping("/applications")
+    public List<AdminPageAnnouncementResponse> getAllAnnouncements(){
+        return announcementServiceImpl.getAllAnnouncements();
+    }
+
+    @GetMapping("/find/{id}")
```
Comment (beksultancs) [inline src/main/java/kg/airbnb/airbnb/apis/AdminApi.java:25]: find what?

File: src/main/java/kg/airbnb/airbnb/apis/AdminApi.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    public AdminPageAnnouncementResponse findAnnouncementById(@PathVariable Long id){
+        return announcementServiceImpl.findAnnouncementById(id);
+    }
+
+    @PutMapping("/announcement/accept/{id}")
```
Comment (beksultancs) [inline src/main/java/kg/airbnb/airbnb/apis/AdminApi.java:30]: Just id is unclear!
You must define like {announcementId}

File: src/main/java/kg/airbnb/airbnb/apis/AdminApi.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+        return announcementServiceImpl.acceptAnnouncement(id);
+    }
+
+    @PutMapping("/announcement/reject/{id}")
+    public SimpleResponse rejectAnnouncement(@PathVariable Long id, @RequestBody AnnouncementRejectRequest announcementRejectRequest){
+        return announcementServiceImpl.rejectAnnouncement(id,announcementRejectRequest);
+    }
+
+    @DeleteMapping("/announcement/delete/{id}")
+    public SimpleResponse deleteAnnouncement(@PathVariable Long id, @RequestBody AnnouncementRejectRequest announcementRejectRequest) {
```
Comment (beksultancs) [inline src/main/java/kg/airbnb/airbnb/apis/AdminApi.java:41]: Do you need a reason when you're delete an announcement?

File: src/main/java/kg/airbnb/airbnb/services/AnnouncementServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+            simpleResponse.setStatus("DELETED");
+            simpleResponse.setMessage(announcementRejectRequest.getMessage());
+            return simpleResponse;
+        }else{
+            throw new ForbiddenException("Only admin can access this page!");
+        }
+
+    }
+
+    private User getAuthenticatedUser() {
```
Comment (beksultancs) [inline src/main/java/kg/airbnb/airbnb/services/AnnouncementServiceImpl.java:115]: I like this code!

**Review by beksultancs — COMMENTED** [review]
almost approved* but you have some moments!

**Comment by beksultancs** [comment]
You have a lot of conflicts here* pull main branch at first and push again!

File: src/main/java/kg/airbnb/airbnb/apis/AdminApi.java (hunk 3 above)
Comment (AltynbekMamasadykov) [reply src/main/java/kg/airbnb/airbnb/apis/AdminApi.java:41]: I'm not sure if we need reason to delete, but, I did not understand why do we need reject message, we not saving it anywhere, how will user who's announcement deleted will see this reject message?

**Comment by AltynbekMamasadykov** [comment]
conflicts handled, ready to merge Sir!

**Review by beksultancs — APPROVED** [review]
Approved*

**Pull request merged**